consensus/spos/bls/benchmark_test.go

```diff
@@ -0,0 +1,238 @@
+
+	sr := *initSubroundEndRoundWithContainer400Sig(container, &statusHandler.AppStatusHandlerStub{}, consensusState, &mock2.ThrottlerStub{})
+	for i := 0; i < len(sr.ConsensusGroup()); i++ {
+		_, err := sr.SigningHandler().CreateSignatureShareForPublicKey(dataToBeSigned, uint16(i), sr.EnableEpochsHandler().GetCurrentEpoch(), []byte(keys[i]))
+		require.Nil(b, err)
+		_ = sr.SetJobDone(keys[i], bls.SrSignature, true)
+	}
+	for i := 0; i < b.N; i++ {
+		b.StartTimer()
+		invalidSigners, err := sr.VerifyNodesOnAggSigFail()
+		b.StopTimer()
+		require.Nil(b, err)
+		require.NotNil(b, invalidSigners)
+	}
+}
+
+func BenchmarkSubroundEndRound_VerifyNodesOnAggSigFailTimeParallelNoThrottle(b *testing.B) {
+	b.ResetTimer()
+	b.StopTimer()
+	container := mock.InitConsensusCore()
+	enableEpochsHandler := &enableEpochsHandlerMock.EnableEpochsHandlerStub{
+		IsFlagEnabledInEpochCalled: func(flag core.EnableEpochFlag, epoch uint32) bool {
+			return flag == common.EquivalentMessagesFlag
+		},
+	}
+	container.SetEnableEpochsHandler(enableEpochsHandler)
+	llSigner := &multisig2.BlsMultiSignerKOSK{}
+	suite := mcl.NewSuiteBLS12()
+	kg := signing.NewKeyGenerator(suite)
+	mapKeys := make(map[string]crypto2.PrivateKey)
+
+	for i := uint16(0); i < 400; i++ {
+		sk, pk := kg.GeneratePair()
+
+		pubKey, _ := pk.ToByteArray()
+		mapKeys[string(pubKey)] = sk
+	}
+
+	multiSigHandler, _ := multisig.NewBLSMultisig(llSigner, kg)
+
+	keysHandlerMock := &testscommon.KeysHandlerStub{
+		GetHandledPrivateKeyCalled: func(pkBytes []byte) crypto2.PrivateKey {
+			return mapKeys[string(pkBytes)]
+		},
+	}
+	keys := createListFromMapKeys(mapKeys)
+	args := crypto.ArgsSigningHandler{
+		PubKeys: keys,
+		MultiSignerContainer: &cryptoMocks.MultiSignerContainerStub{
+			GetMultiSignerCalled: func(epoch uint32) (crypto2.MultiSigner, error) {
+				return multiSigHandler, nil
+			},
+		},
+		SingleSigner: &cryptoMocks.SingleSignerStub{},
+		KeyGenerator: kg,
+		KeysHandler:  keysHandlerMock,
+	}
+
+	signingHandler, err := crypto.NewSigningHandler(args)
+	require.Nil(b, err)
+
+	container.SetSigningHandler(signingHandler)
+	consensusState := initConsensusStateWithArgs(keysHandlerMock, keys)
+	dataToBeSigned := []byte("message")
+	consensusState.Data = dataToBeSigned
+
+	sr := *initSubroundEndRoundWithContainer400Sig(container, &statusHandler.AppStatusHandlerStub{}, consensusState, &mock2.ThrottlerStub{})
+	for i := 0; i < len(sr.ConsensusGroup()); i++ {
+		_, err := sr.SigningHandler().CreateSignatureShareForPublicKey(dataToBeSigned, uint16(i), sr.EnableEpochsHandler().GetCurrentEpoch(), []byte(keys[i]))
+		require.Nil(b, err)
+		_ = sr.SetJobDone(keys[i], bls.SrSignature, true)
+	}
+	for i := 0; i < b.N; i++ {
+		b.StartTimer()
+		invalidSigners, err := sr.VerifyNodesOnAggSigFailAux()
+		b.StopTimer()
+		require.Nil(b, err)
+		require.NotNil(b, invalidSigners)
+	}
+}
+
+func BenchmarkSubroundEndRound_VerifyNodesOnAggSigFailTimeParallelThrottle(b *testing.B) {
```

Comment:
we would end up only with this benchmark, probably the others need to be removed.